cmake-modules/AzureBuildTargetForCI.cmake

```diff
@@ -32,6 +32,21 @@ macro(create_per_service_target_build_for_sample  service target)
     if(CMAKE_GENERATOR MATCHES "Visual Studio.*")
         SET(binary "Release/${binary}")
     endif()
-    file(APPEND ${CMAKE_BINARY_DIR}/${service}-samples.txt "${CMAKE_CURRENT_BINARY_DIR}/${binary}\n")
-    
+
+    set(RUN_SAMPLE TRUE)
+    if(${ARGC} GREATER 2)
+      set(extra_args "${ARGN}")
+      foreach(arg IN LISTS extra_args)
+        if (${arg} MATCHES "DISABLE_RUN")
+          set(RUN_SAMPLE FALSE)
+          message("Disabling sample execution for ${service}/${binary}")
+        else()
+          message(FATAL_ERROR "Unknown extra argument: ${arg}")
+        endif()
```

Comment:
If we find the first arg match for DISABLE_RUN, should we break out of the loop after setting RUN_SAMPLE to false? What's the point of keep checking all the extra_args?

Reply:
If we later add more optional parameters, then the break would have to be removed. This way we don't have to do that.

And since there's currently only one additional parameter, it doesn't matter.

Reply:
> If we later add more optional parameters, then the break would have to be removed.

Why?

Even if there were more optional parameters, we iterate through all of them, find the presence of DISABLE_RUN, set RUN_SAMPLE to false and break. We don't need to go through all params, after we find the one we need.
```Cmake
        foreach(arg IN LISTS extra_args)
            if (${arg} MATCHES "DISABLE_RUN")
                set(RUN_SAMPLE FALSE)
                message("Disabling sample execution for ${service}/${binary}")
                break() // No harm in breaking here, no need to check further
            else()
                message(FATAL_ERROR "Unknown extra argument: ${arg}")
            endif()
        endforeach()
```

Reply:
If there are 4 optional parameters, you cannot break after finding DISABLE_RUN, because it means we stop iterating over the parameters.

The idea of this loop is that it processes the optional parameters, collecting the state of each of them. Once the parameters have been processed, we deal with each of the options one at a time.

That's why the RUN_SAMPLE thingy is handled the way it is.